crypto/fipsmodule/rand/snapsafe_detect_test.cc

```diff
@@ -0,0 +1,119 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/base.h>
+#include <cstdint>
+
+#include <gtest/gtest.h>
+#include "snapsafe_detect.h"
+
+#if defined(OPENSSL_LINUX) && defined(AWSLC_SNAPSAFE_TESTING)
+#include <fcntl.h>
+#include <cstring>
+#include <sys/mman.h>
+
+#define NUMBER_OF_TEST_VALUES 5
+
+typedef struct sgn_test_s {
+  void* addr;
+  size_t pgsize;
+} sgn_test_s;
+
+int init_sgn_file(sgn_test_s* sgn_test);
+int init_sgn_file(sgn_test_s* sgn_test) {
+  const char* sgc_file_path = AWSLC_SYSGENID_PATH;
+  int fd_sgn = open(sgc_file_path, O_CREAT | O_RDWR | O_APPEND, S_IRUSR | S_IWUSR);
+  if (fd_sgn == -1) {
```

Comment:
can use the `ASSERT_` and `EXPECT_` in these utility functions as well.

Reply:
But if we do that we would be unable to cleanup resources, e.g., close the file. I could use `FAIL()` instead of returning 0?

Reply:
OK that makes sense.